### PR TITLE
Renaming for code review

### PR DESCRIPTION
**Pull request by brunokrebs**

### CODE REVIEW

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+---
+layout: post
+title: "Build Your First Swift Web App with Vapor"
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:3]: I would change this article's title to "Server Side Swift App with Vapor and Droplet". I think it would get more attention and would perform better from a SEO perspective.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+- vapor
+- swift
+- server-side
+- backend
+- rest
+---
+
+---
+
+**TL;DR**: The Swift language is growing in popularity not only for iOS apps, but also for server-side applications. In this article, we'll use Vapor framework to build a web app.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:25]: Then I would say "we'll use Vapor and Droplet to build a web app."

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+---
+
+Since it was introduced in mid-2014, Swift language has seen [stunning growth in popularity](http://redmonk.com/sogrady/2016/07/20/language-rankings-6-16/). When the language became [open source](https://developer.apple.com/swift/blog/?id=34) and available on linux in late-2015, server-side applications started to gain popularity as well.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:29]: Capital L for Linux.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+Recently, Apple announced the creation of [Swift Server APIs working group](https://swift.org/blog/server-api-workgroup/). They invite participants in the community to provide new Swift APIs focused on server-side needs (as networking, security and HTTP/WebSocket parsing). This way, it will be possible to write pure Swift server-side frameworks, without needing to rely on C libraries.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:31]: They invite => They invited

Comment (brunokrebs) [reply server-side-swift-app-with-vapor-and-droplet.md:31]: I would delete the paragraph below (that starts with "The work group steering team and stakeholders") and would rephrase the second sentence here to say something like:

IBM Kitura, Vapor, Zewo and Perfect were invited by Apple to focus on server-side needs like networking, security and HTTP/WebSocket parsing. This way, they made possible to write pure Swift server-side frameworks, without needing to rely on C libraries.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+---
+
+Since it was introduced in mid-2014, Swift language has seen [stunning growth in popularity](http://redmonk.com/sogrady/2016/07/20/language-rankings-6-16/). When the language became [open source](https://developer.apple.com/swift/blog/?id=34) and available on linux in late-2015, server-side applications started to gain popularity as well.
+
+Recently, Apple announced the creation of [Swift Server APIs working group](https://swift.org/blog/server-api-workgroup/). They invite participants in the community to provide new Swift APIs focused on server-side needs (as networking, security and HTTP/WebSocket parsing). This way, it will be possible to write pure Swift server-side frameworks, without needing to rely on C libraries. 
+
+The work group steering team and stakeholders consists of people involved in the hottest Swift server-side frameworks at the moment: [IBM Kitura](https://github.com/IBM-Swift/Kitura), [Vapor](https://github.com/vapor/vapor), [Zewo](https://github.com/Zewo/Zewo) and [Perfect](https://github.com/PerfectlySoft/Perfect).
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:33]: I would delete this.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+In this tutorial, we'll learn how to set up Vapor framework, learn its essential concepts, and build a simple web application for creating and retrieving contacts.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:35]: Add a link to Vapor framework.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+## Prerequisites
+
+You need a machine [able to run Swift 3](https://swift.org/download/).
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:39]: Elaborate this a little bit more. Add info like what kind of machine (OS?) is currently supported and if they expect to support additional architectures in the future.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+## Environment setup
+
+First of all, we need to [download and install Swift 3](https://swift.org/download/) (the steps are different for Ubuntu and macOS). After that, it's necessary to install the Vapor Toolbox, a command line interface for common Vapor tasks, as `build` and `serve`. Run the following script to install the Toolbox.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:43]: Add a link to Vapor Toolbox.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+Recently, Apple announced the creation of [Swift Server APIs working group](https://swift.org/blog/server-api-workgroup/). They invite participants in the community to provide new Swift APIs focused on server-side needs (as networking, security and HTTP/WebSocket parsing). This way, it will be possible to write pure Swift server-side frameworks, without needing to rely on C libraries. 
+
+The work group steering team and stakeholders consists of people involved in the hottest Swift server-side frameworks at the moment: [IBM Kitura](https://github.com/IBM-Swift/Kitura), [Vapor](https://github.com/vapor/vapor), [Zewo](https://github.com/Zewo/Zewo) and [Perfect](https://github.com/PerfectlySoft/Perfect).
+
+In this tutorial, we'll learn how to set up Vapor framework, learn its essential concepts, and build a simple web application for creating and retrieving contacts.
+
+## Prerequisites
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:37]: Change to "## Swift Server Side Prerequisites" or something like that. Our SEO expert instructed us to have some headings with the keyword (swift in this case I guess)

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+You need a machine [able to run Swift 3](https://swift.org/download/).
+
+## Environment setup
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:41]: For the same reason, change this to "## Setting up Swift and Vapor"

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+To verify if the installation was successful, run `vapor --help`, which should show a list of commands. To update the toolbox, just run `vapor self update`.
+
+To make sure everything is running fine, let's build a Hello World app.
+
+### Hello World App
+
+First of all, let's create the new Vapor project.
+
+`vapor new HelloWorld`
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:55]: Change to a code block (three back-ticks). It will look better.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+
+Vapor will generate a folder structure with some files. Vapor command line interface uses [Swift Package Manager](https://swift.org/package-manager/) to manage the dependencies.
+
+If you are using macOS and want to edit the project on Xcode, run `vapor xcode -y` in the HelloWorld subdirectory to generate and open an Xcode project.
+
+Look for the main.swift file and place the following code there.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:61]: Format main.swift as inline code (single back-tick).

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+> Note: on version 1.1, Vapor already generates this code when you create a new project. But it's here just in case things change, considering that Vapor is in a fast development pace.
+
+After that, run the two following commands.
+
+```
+vapor build
+vapor run serve
+```
+
+And it should be done. Now vapor is serving its HelloWorld page on localhost:8080. To check that things are working out, just open your browser and type `localhost:8080`.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:89]: ... HelloWorld page on `http://localhost:8080` ... and type and type `http://localhost:8080`.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+
+
+## Bulding a Contacts App
+
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:93]: Elaborate a little here. Say something like "Now that we have our server-side Swift environment setup working with Vapor we are going to start creating our *Contacts App*. This application will have two features: blablabla and something else."

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+### Playing with Droplet
+
+Droplet is a container that makes it easy to use many of Vapor's features. It's responsible for starting and running the server, set up environments, register routes, and more. It's necessary to import Vapor to initialize and run a droplet.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:96]: Add a link to Droplet.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+
+```swift
+import Vapor
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:99]: So you have to import it, but you don't need to use it (directly)? If so, that is what a call weird.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+
+Let's now register a get request at localhost:8080/hello.
+
+```swift
+drop.get("hello") { request in
+    return "Hello World!"
+}
+```
+
+The first parameter is the endpoint, and the closure returns the response of the request. The return must be of a type who conforms to the `ResponseRepresentable` protocol. Strings and JSON already conform to this protocol, but it's also possible to create a custom response.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:116]: "The return must be of a" => "The returned value must be of a"

Comment (brunokrebs) [reply server-side-swift-app-with-vapor-and-droplet.md:116]: "and the closure returns the response of the request" => "and the closure returns a response to the request"

Comment (brunokrebs) [reply server-side-swift-app-with-vapor-and-droplet.md:116]: "to create a custom response" <= would you add a link to some doc that shows how to do this?

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 18]
```diff
@@ -0,0 +1,172 @@
...
+
+```swift
+drop.get("hello") { request in
+	return Response(status: .ok, headers: ["Content-Type": "text/plain"], body: "Hello, World!")
+}
+```
+
+It's also pretty straightforward to create a post request and retrieve sent JSON data.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:124]: Actually what you are creating is an endpoint that responds to POST requests. And this endpoint returns data in JSON format.

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 19]
```diff
@@ -0,0 +1,172 @@
...
+
+```swift
+drop.post("contacts", "create") { request in
+    return try JSON(node: [
+        "response": request.data["name"]?.string
+        ])
+}
+```
+
+This register a POST HTTP endpoint at localhost:8080/contacts/create and returns a JSON response. It gets the data sent as JSON in the request for the key test and retrieves its string value on `request.data["test"]?.string`.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:134]: Exactly, the sentence above the code is misleading.

Comment (brunokrebs) [reply server-side-swift-app-with-vapor-and-droplet.md:134]: localhost:8080/contacts/create => `http://localhost:8080/contacts/create`

Comment (brunokrebs) [reply server-side-swift-app-with-vapor-and-droplet.md:134]: " for the key test and " <= I don't understand... 'key test' what do you mean?

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 20]
```diff
@@ -0,0 +1,172 @@
...
+
+This endpoint is easy to test with cURL:
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:136]: "This endpoint is easy to test with cURL" => "To test this endpoint we can use a program like [Postman](https://www.getpostman.com/) or cURL:

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 21]
```diff
@@ -0,0 +1,172 @@
...
+
+`curl -i -H "Accept: application/json" "localhost:8080/contacts/create {"name":"John"}"`
+
+
+Don't forget to always run `vapor build` and `vapor run serve` before testing on the browser or with cURL.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:141]: "Don't forget to always run " => remove always

Comment (brunokrebs) [reply server-side-swift-app-with-vapor-and-droplet.md:141]: "before testing on the browser or with cURL." => "before testing it"

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 22]
```diff
@@ -0,0 +1,172 @@
...
+
+
+### Database integration
+
+Let's make the database integration with MySQL. First of all, it's necessary to install MySQL and start it.
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:146]: Add a link to MySQL

File: server-side-swift-app-with-vapor-and-droplet.md [hunk 23]
```diff
@@ -0,0 +1,172 @@
...
+```
+
+On macOS:
+
+```sh
+brew install mysql
+brew link mysql
+mysql.server start
+```
+> In case you don't have Homebrew installed, follow the instructions at http://brew.sh/
```
Comment (brunokrebs) [inline server-side-swift-app-with-vapor-and-droplet.md:164]: "follow the instructions at [Homebrew's homepage](http://brew.sh/).

File: server-side-swift-app-with-vapor-and-droplet.md (hunk 1 above)
Comment (rsacchi) [reply server-side-swift-app-with-vapor-and-droplet.md:3]: Right. Can you confirm if it's ok with tony?

File: server-side-swift-app-with-vapor-and-droplet.md (hunk 7 above)
Comment (rsacchi) [reply server-side-swift-app-with-vapor-and-droplet.md:39]: This is all described on the linked Swift website. Do you think it's necessary to repeat in the article? Maybe something quite straightforward like 

You need a machine running Mac OS X 10.11+ or Linux. Apple builds and tests binaries for Ubuntu 14.04 and 16.04 and 16.10, but it's possible to build the language from source in other Linux distributions.

File: server-side-swift-app-with-vapor-and-droplet.md (hunk 16 above)
Comment (rsacchi) [reply server-side-swift-app-with-vapor-and-droplet.md:99]: In Swift, you don't import files - you import modules.
The Droplet class is part of the Vapor module. Better now? :)

File: server-side-swift-app-with-vapor-and-droplet.md (hunk 17 above)
Comment (rsacchi) [reply server-side-swift-app-with-vapor-and-droplet.md:116]: Yep!

File: server-side-swift-app-with-vapor-and-droplet.md (hunk 18 above)
Comment (rsacchi) [reply server-side-swift-app-with-vapor-and-droplet.md:124]: Good point. Thanks :)

File: server-side-swift-app-with-vapor-and-droplet.md (hunk 19 above)
Comment (rsacchi) [reply server-side-swift-app-with-vapor-and-droplet.md:134]: the key "test" on the JSON object.

Comment (rsacchi) [reply server-side-swift-app-with-vapor-and-droplet.md:134]: Which sentence is misleading? The one you mentioned in the prior comment - "It's also pretty straightforward to create a post request and retrieve sent JSON data." ?

File: server-side-swift-app-with-vapor-and-droplet.md (hunk 7 above)
Comment (brunokrebs) [reply server-side-swift-app-with-vapor-and-droplet.md:39]: Yes, that is much better. It gives more context.

File: server-side-swift-app-with-vapor-and-droplet.md (hunk 19 above)
Comment (brunokrebs) [reply server-side-swift-app-with-vapor-and-droplet.md:134]: "Which sentence is misleading? The one you mentioned in the prior comment?" <= exactly

File: server-side-swift-app-with-vapor-and-droplet.md (hunk 6 above)
Comment (rsacchi) [reply server-side-swift-app-with-vapor-and-droplet.md:35]: Added on the prior paragraph
